scenes/enumerated/problem_12189/problem_12189.py

```diff
@@ -0,0 +1,170 @@
+        x2_one_part_value_brace = Brace(VGroup(x3_one_part[1], x3_one_part[2]), UP)
+        x2_one_part_value_brace_tex = x2_one_part_value_brace.get_tex("18")
+        x2_one_part_value = MathTex("13+5=18").to_edge(UL, buff=1.1).shift(DOWN*.3)
+        self.play(
+            Write(x2_one_part_value),
+            run_time=2
+        )
+        self.wait()
+        self.play(
+            GrowFromCenter(x2_one_part_value_brace),
+            Write(x2_one_part_value_brace_tex)
+        )
+        self.wait()
+        one_part_value = MathTex("18:2=9").next_to(x2_one_part_value, DOWN, buff=.7, aligned_edge=LEFT)
+        self.play(Write(one_part_value), run_time=2)
+        one_part_label = MathTex("9").align_to(one_part_value, UR)
+        one_part_label_copy = one_part_label.copy()
+        self.play(VGroup(x2_one_part_value_brace_tex, x2_one_part_value_brace).animate.shift(UP*.7))
+        self.play(
+            one_part_label.animate.next_to(x3_one_part[1], UP),
+            one_part_label_copy.animate.next_to(x3_one_part[2], UP)
+        )
+        self.wait()
+
+        armen_one_part.set_label(MathTex("9"))
+        armen_one_part_label = armen_one_part.update_label_pos()
+        vazgen_one_part.set_label(MathTex("9"))
+        vazgen_one_part_label = vazgen_one_part.update_label_pos()
+        self.play(
+            FadeIn(
+                armen_one_part_label,
+                vazgen_one_part_label
+            )
+        )
+        self.wait()
+        self.play(
+            VGroup(
+                vazgen_segment_13,
+                vazgen_segment_13_label
+            ).animate.shift(LEFT),
+            FadeOut(
+                segment_5,
+                segment_5_label,
+                *[one_part for one_part in x3_one_part],
+                one_part_label,
+                one_part_label_copy,
+                x2_one_part_value_brace_tex, x2_one_part_value_brace
+            )
+        )
+        self.wait()
+        vazgen_mushrooms = MathTex("9+13=22").next_to(x2_one_part_value, buff=2)
+        self.play(Write(vazgen_mushrooms), run_time=2)
+        vazgen_brace = Brace(VGroup(vazgen_one_part, vazgen_segment_13), UP, buff=.4)
+        vazgen_brace_label = MathTex("22").align_to(vazgen_mushrooms, UR)
+        self.play(
+            GrowFromCenter(vazgen_brace),
+            vazgen_brace_label.animate.next_to(vazgen_brace, UP)
+        )
+        self.wait()
+
+        together_value = MathTex("22+9=31").next_to(vazgen_mushrooms, DOWN, buff=.7, aligned_edge=LEFT)
+        self.play(Write(together_value), run_time=2)
+        together_brace = Brace(VGroup(armen_one_part, vazgen_one_part), RIGHT, buff=4.5)
+        together_brace_label = MathTex("31").align_to(together_value, UR)
+        self.play(
+            GrowFromCenter(together_brace),
+            together_brace_label.animate.next_to(together_brace)
+        )
+
+        self.wait(2)
+
+    def get_part(self, value, label = None, color = WHITE):
```

Comment:
Can't have this function in some helpers in `lib` or `scenes/enumerated` or just anywhere, but not in every class?